src/Backend/WASMWrapper.js

```diff
@@ -52,10 +51,9 @@ export default class WASMWrapper {
         const buf = wasm._malloc(this.#ionsSize * Int32Array.BYTES_PER_ELEMENT)
         const heap = new Uint32Array(wasm.HEAP32.buffer, buf, this.#ionsSize)
 
-        wasm.ccall("dedx_get_ion_list", null, ['number', 'number'], [program, heap.byteOffset])
+        wasm.ccall("dedx_get_ion_list", null, ['number', 'number'], [program - 1, heap.byteOffset])
```

Comment:
Why `-1` is needed? I believe both JS and C are numbering array starting from 0.

If `program` is not an index in the array then what is it ?
Is it by a chance number of item in drop down list which is numbered from 1 ?

If that is the case, then it should be renamed to something like `program_item_number`.
Also - a good comment is needed which explains how numbering is done

Reply:
Program is the program id supplied by libdedx. I with i could tell how were the ids chosen, but i didn't see any explaination anywhere in the libdex.
![image](https://user-images.githubusercontent.com/39342674/146974420-77925b7d-4bed-4940-98a0-2d0e06dec54c.png)

Reply:
Apart from 2 comments "/* ok, this is a bit overkill... */

/* Hey Jakob, whats the purpose with  "" placeholder ? */"
Which i agree with

Reply:
And -1 is the result of the fact that we're using the libdedx program ID instead of array index so now we have to go back one because ASTAR has the ID of 1 and its ions and materials are in 0-indexed arrays

Reply:
Quick update on the -1: So from the beggining we were using all ions as the dedicated set of ions for ASTAR so when it started working correctly i assumed its broken so i added the -1